### PR TITLE
Introduce "show" method in slicer

### PR DESCRIPTION
**Pull request by AlexandreAbraham**
Introducing this method simplify the code because we don't have to import matplotlib most of the time.

### CODE REVIEW

**Comment by lesteve** [comment]
Pfff controversial. It seems fine to me to require users to use plt.show() ...

Also you are kind of lying that if you plot multiple displays, calling .show on one will show all of them.

**Comment by GaelVaroquaux** [comment]
:+1:

Can you please add it to plotting.rst, in a box somewhere.

**Comment by lesteve** [comment]
Also it goes against the fact that non-advanced users shouldn't have to use the object returned by the plot_* functions.

**Comment by GaelVaroquaux** [comment]
> Pfff controversial. It seems fine to me to require users to use plt.show() ...

The less 'imports' we need to teach to a beginner, the easier it is.

That said, we need to document (in docstrings and narrative docs) that it
is exactly matplotlib.show.

**Comment by lesteve** [comment]
If we really think `import matplotlib.pyplot as plt; plt.show()` is out of reach why not add a `show` function in `nilearn.plotting` ?

**Comment by AlexandreAbraham** [comment]
@lesteve proposition seems reasonable to me.

**Comment by AlexandreAbraham** [comment]
@GaelVaroquaux Could you answer this one please? @lesteve suggested that we could put the `show` function directly in the module, since it's not related to a particular plot. I think it's a good suggestion.

**Comment by GaelVaroquaux** [comment]
> @GaelVaroquaux Could you answer this one please? @lesteve suggested that we
> could put the show function directly in the module, since it's not related to a
> particular plot. I think it's a good suggestion.

I think so too. Sorry, I said it orally yesterday, but orally is not
good.

**Comment by GaelVaroquaux** [comment]
Can I haz documentation in plotting.rst plz

**Comment by GaelVaroquaux** [comment]
This PR is almost ready. It needs just narrative documentation.

**Comment by lesteve** [comment]
This one needs a rebase.

File: examples/connectivity/plot_canica_resting_state.py [hunk 1]
```diff
@@ -47,10 +47,8 @@
 components_img.to_filename('canica_resting_state.nii.gz')
 
 ### Visualize the results #####################################################
-
-import matplotlib.pyplot as plt
-from nilearn.plotting import plot_prob_atlas
```
Comment (lesteve) [inline examples/connectivity/plot_canica_resting_state.py]: Boo, why did you remove the plot_prob_atlas import?

File: examples/connectivity/plot_canica_resting_state.py [hunk 2]
```diff
@@ -47,10 +47,8 @@
 components_img.to_filename('canica_resting_state.nii.gz')
 
 ### Visualize the results #####################################################
-
```
Comment (lesteve) [inline examples/connectivity/plot_canica_resting_state.py]: Also you added back this comment ... which was removed at one point

File: examples/manipulating_visualizing/plot_demo_glass_brain_extensive.py [hunk 3]
```diff
@@ -32,8 +32,7 @@
 plotting.plot_glass_brain(localizer_tmap_filename, threshold=0, colorbar=True,
                           plot_abs=False)
 
-plotting.plot_glass_brain(localizer_tmap_filename, threshold=3, colorbar=True,
-                          plot_abs=False)
+plotting.plot_glass_brain(localizer_tmap_filename, threshold=3,
+                                    colorbar=True, plot_abs=False)
```
Comment (lesteve) [inline examples/manipulating_visualizing/plot_demo_glass_brain_extensive.py]: PEP8

File: examples/connectivity/plot_canica_resting_state.py (hunk 2 above)
Comment (AlexandreAbraham) [inline examples/connectivity/plot_canica_resting_state.py]: Sorry, bad merge.

File: examples/connectivity/plot_canica_resting_state.py (hunk 1 above)
Comment (AlexandreAbraham) [inline examples/connectivity/plot_canica_resting_state.py]: Bad merge, fixed.

File: examples/manipulating_visualizing/plot_demo_glass_brain_extensive.py (hunk 3 above)
Comment (AlexandreAbraham) [inline examples/manipulating_visualizing/plot_demo_glass_brain_extensive.py]: Weird. But fixed.

**Comment by lesteve** [comment]
LGTM, merging.

**Comment by GaelVaroquaux** [comment]
There are layout problems:

https://circle-artifacts.com/gh/nilearn/nilearn/188/artifacts/0/home/ubuntu/nilearn/doc/_build/html/manipulating_visualizing/plotting.html

In addition, you want to phrase the problem the other way around: first
introduce 'show', and then say it's matplotlib's show:

I'll rephrase it directly in master (sudo make me a sandwich version of
commenting on a PR :$)

**Comment by AlexandreAbraham** [comment]
Thanks, I appreciate that. Since I switched priorites, I did not want to come back on this one.
